src/sentry/snuba/metrics_performance.py

```diff
@@ -527,9 +529,12 @@ def top_events_timeseries(
                     else item["data"]
                 ),
                 "order": item["order"],
-                # One of the queries in the builder has required, thus, we mark all of them
-                # This could mislead downstream consumers of the meta data
-                "meta": {"isMetricsExtractedData": top_events_builder.use_on_demand},
+                "meta": {
+                    # One of the queries in the builder has required, thus, we mark all of them
+                    # This could mislead downstream consumers of the meta data
+                    **{"isMetricsExtractedData": top_events_builder.use_on_demand},
+                    **result["meta"],
+                },
```

Comment:
No need unpack a dict here

```suggestion
                "meta": {
                    # One of the queries in the builder has required, thus, we mark all of them
                    # This could mislead downstream consumers of the meta data
                    "isMetricsExtractedData": top_events_builder.use_on_demand,
                    **result["meta"],
                },
```